eq_cir_management_ui/__init__.py

```diff
@@ -0,0 +1,116 @@
+"""Flask application factory for the EQ CIR Management UI."""
+
+import json
+import logging
+import os
+from pathlib import Path
+
+from flask import Flask
+from flask_talisman import Talisman
+from jinja2 import ChainableUndefined, FileSystemLoader
+
+from eq_cir_management_ui.config.config import DefaultConfig
+from eq_cir_management_ui.errors.routes import errors_blueprint
+from eq_cir_management_ui.main.routes import main_blueprint
+from eq_cir_management_ui.utils.routes import utils_blueprint
+
+logger = logging.getLogger()
+
+talisman = Talisman()
+
+
+def create_app(app_config: type[DefaultConfig]) -> Flask:
+    """Flask application factory, used to isolate the instance of the Flask application.
+    See https://flask.palletsprojects.com/en/2.2.x/patterns/appfactories/ .
+    """
+    app = Flask(__name__)
+
+    app.config.from_object(app_config)
+    app.static_folder = str(Path("../static"))
+
+    app.register_blueprint(main_blueprint)
+    app.register_blueprint(errors_blueprint)
+    app.register_blueprint(utils_blueprint)
+
+    jinja_config(app)
+    design_system_config()
+    configure_secure_headers(app)
+
+    return app
+
+
+def jinja_config(app: Flask) -> None:
+    """Configuration for the Flask Jinja2 component. Here we provide a custom loader,
+    so we can load from an array of sources.
+
+    :param app: The Flask application.
+    """
+    # loader for local templates and design system component templates
+    file_system_loader = FileSystemLoader([Path("./node_modules/@ons/design-system"), Path("./templates")])
+
+    app.jinja_loader = file_system_loader
+    app.jinja_env.undefined = ChainableUndefined
+
+    # Clean up white space.
+    app.jinja_env.trim_blocks = True
+    app.jinja_env.lstrip_blocks = True
+
+
+def design_system_config() -> None:
+    """Set the version of the design system to an environment variable and add an
+    environment variable filter so environment variables can be read from within
+    Jinja. This enables the design system version to be defined once within the
+    package.json file and then reused throughout the application. Primarily to
+    declare the CSS version to use.
+
+    :param app: The Flask application.
+    """
+    with open(Path("./package.json"), encoding="utf-8") as file:
```

Comment:
Could use an absolute path here instead of relative paths - similar to above?

Reply:
Should probably give my reasoning for this, as I didn't in my original comment sorry - This suggestion was to avoid ambiguity regardless of the current working directory, and ensures that the path always points to the same location (mostly for when running this stuff locally)

Reply:
This is what I had in mind: 
```
with open(Path(__file__).parent.parent / "package.json", encoding="utf-8") as file:
```
Hopefully that should work?

Reply:
Think this should work - think I only had one `.parent` instead of 2.
I'm not sure how this is an absolute path though, if you have to tell it where the file/root is relative to the code.

Reply:
Tried the change, and it works fine, but the tests break I think because the path relative to the file is different with the tests.
As it works for render and tests currently will leave as is.